### PR TITLE
Added docker compose example for AgentQnA deployment on AMD ROCm 

### PR DESCRIPTION
**Pull request by artem-astafev**
## Type of change

- Added docker compose example for AgentQnA deployment on AMD ROCm (TGI service is in use).  
- Added CI tests for Docker compose AgentQnA deploymen example on AMD ROCm systems

## Tests

Go to GenAIExamples/AgentQnA/tests and run test_compose_on_rocm.sh bash script (./test_compose_on_rocm.sh).

### CODE REVIEW

**Review by lvliang-intel — APPROVED** [review]

File: AgentQnA/docker_compose/amd/gpu/rocm/launch_agent_service_tgi_rocm.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Copyright (C) 2024 Advanced Micro Devices, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+WORKPATH=$(dirname "$PWD")/..
+# export WORKDIR=$WORKPATH/../../
+echo "WORKDIR=${WORKPATH}"
```
Comment (dcmiddle) [inline AgentQnA/docker_compose/amd/gpu/rocm/launch_agent_service_tgi_rocm.sh:6]: Do you intend the comment and echo?

File: AgentQnA/docker_compose/amd/gpu/rocm/set_env.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# Copyright (C) 2024 Advanced Micro Devices, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+WORKPATH=$(dirname "$PWD")/..
+# export WORKDIR=$WORKPATH/../../
+echo "WORKDIR=${WORKPATH}"
```
Comment (dcmiddle) [inline AgentQnA/docker_compose/amd/gpu/rocm/set_env.sh:8]: comment and echo?

File: AgentQnA/docker_compose/amd/gpu/rocm/set_env.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+export LLM_MODEL_ID="meta-llama/Meta-Llama-3-8B-Instruct"
+export NUM_SHARDS=4
+export LLM_ENDPOINT_URL="http://${ip_address}:${AGENTQNA_TGI_SERVICE_PORT}"
+export temperature=0.01
+export max_new_tokens=512
+
+# agent related environment variables
+export AGENTQNA_WORKER_AGENT_SERVICE_PORT="9095"
+export TOOLSET_PATH=$WORKDIR/GenAIExamples/AgentQnA/tools/
+echo "TOOLSET_PATH=${TOOLSET_PATH}"
```
Comment (dcmiddle) [inline AgentQnA/docker_compose/amd/gpu/rocm/set_env.sh:28]: intentional echo?

**Comment by dcmiddle** [comment]
please squash your fixup commits

**Comment by artem-astafev** [comment]
@dcmiddle 
> please squash your fixup commits

Is it necessary? If I understand write all commits will be squashed after merge.

**Comment by dcmiddle** [comment]
> @dcmiddle
> 
> > please squash your fixup commits
> 
> Is it necessary? If I understand write all commits will be squashed after merge.

Yes please.
The goal is to arrive at a commit message that explains why a change was made. When the commit history is littered with fixup comments it makes it harder for people to understand the code using tools like git blame.

File: AgentQnA/docker_compose/amd/gpu/rocm/set_env.sh (hunk 3 above)
Comment (dcmiddle) [reply AgentQnA/docker_compose/amd/gpu/rocm/set_env.sh:26]: Please remove debug echo's unless they have some long term logging purpose.

File: AgentQnA/docker_compose/amd/gpu/rocm/set_env.sh (hunk 2 above)
Comment (artem-astafev) [reply AgentQnA/docker_compose/amd/gpu/rocm/set_env.sh]: Removed

File: AgentQnA/docker_compose/amd/gpu/rocm/set_env.sh (hunk 3 above)
Comment (artem-astafev) [reply AgentQnA/docker_compose/amd/gpu/rocm/set_env.sh]: Removed

**Comment by artem-astafev** [comment]
> > @dcmiddle
> > > please squash your fixup commits
> > 
> > 
> > Is it necessary? If I understand write all commits will be squashed after merge.
> 
> Yes please. The goal is to arrive at a commit message that explains why a change was made. When the commit history is littered with fixup comments it makes it harder for people to understand the code using tools like git blame.

@dcmiddle  Done, I've squash all changes into one commit.

**Pull request merged**